meson.build

```diff
@@ -199,39 +199,36 @@ endif
 # CoreText
 coretext = false
 if not get_option('coretext').disabled()
-    coretext_dep = dependency(
+    appservices_dep = dependency(
         'appleframeworks',
-        modules: ['CoreText', 'CoreFoundation'],
+        modules: ['ApplicationServices', 'CoreFoundation'],
         required: false,
     )
-
-    if (
-        coretext_dep.found()
-        and cc.has_type(
-            'CTRunRef',
-            prefix: '#include <CoreText/CoreText.h>',
-            dependencies: coretext_dep,
-        )
+    # this intentionally includes a leading newline
+    coretext_check = '''
+        int main(void) {
+            CTFontDescriptorCopyAttribute(NULL, kCTFontNameAttribute);
+            return 0;
+        }
+    '''
+    appservices_snippet = (
+        '#include <ApplicationServices/ApplicationServices.h>' + coretext_check
     )
-        deps += coretext_dep
+
+    if appservices_dep.found() and cc.compiles(appservices_snippet)
+        deps += appservices_dep
```

Comment:
hmm, autoconf has [`AC_CHECK_TYPE(S)`](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Generic-Types.html) in all remotely relevant versions

If both checks are equivalent *(i feel like this was discussed before at some point)* i wonder if we could make the check more concise with it. I guess one potential drawback is `AC_CHECK_TYPE(S)` log output disrupting our custom CoreText check output?

anyway, that’s just a side note until it actually changes and if it’s simple enough to do, using the same check in both is ideal